src/Libraries/CoreNodes/List.cs

```diff
@@ -52,6 +52,375 @@ public static bool ContainsItem(IList list, object item)
+        }
+
+        /// <summary>
+        ///     Inserts an element into a list at specified index.
+        /// </summary>
+        /// <param name="list">The list the element will be inserted to.</param>
+        /// <param name="element">The element to be inserted.</param>
+        /// <param name="index">Specifies the location in the list of the element to be inserted.</param>
+        /// <returns name="list">The list with the element inserted.</returns>
+        /// <search>insert,add</search>
+        public static IList Insert(IList list, [ArbitraryDimensionArrayImport] object element, int index)
+        {
+            list.Insert(index, element);
+            return list;
+        }
+
+        /// <summary>
+        ///     Reorders the input list based on the given list of indices.
+        /// </summary>
+        /// <param name="list">The list to be reordered.</param>
+        /// <param name="indice">The indices used to reorder the items in the list.</param>
+        /// <returns name="list">The reordered list.</returns>
+        /// <search>reorder,index,indices</search>
+        public static IList Reorder(IList list, IList indice)
+        {
+            // Note: slightly different behaviour from Builtin - invalid input indices will be ignored, 
+            // and will return a list instead of null if the indices are invalid
+            List<object> newList = new List<object>();
+            for (int i = 0; i < indice.Count; i++)
+            {
+                int index;
+                if ((int.TryParse(indice[i].ToString(), out index) && (index < list.Count)))
+                {
+                    newList.Add(list[index]);
+                }
+            }
+            return newList;
+        }
+
+        /// <summary>
+        ///     Sorts a list by the items and return their indices.
+        /// </summary>
+        /// <param name="list">The list of items to be sorted.</param>
+        /// <returns name="newList">The indices of the items in the sorted list.</returns>
+        public static IList SortIndexByValue(List<double> list)
+        {
+            List<Tuple<int, double>> tupleList = new List<Tuple<int, double>>();
+            for (int i = 0; i < list.Count; i++)
+            {
+                tupleList.Add(new Tuple<int, double>(i, list[i]));
+            }
+            tupleList = tupleList.OrderBy(x => x.Item2).ToList();
+            List<int> newList = new List<int>();
+            foreach (Tuple<int, double> x in tupleList)
+            {
+                newList.Add(x.Item1);
+            }
+            return newList;
+        }
+        
+        /// <summary>
+        ///     Returns multidimentional list according to the rank of the input list.
+        /// </summary>
+        /// <param name="list">The list whose depth is to be normalized.</param>
+        /// <returns name="list">The list with the normalized rank.</returns>
+        public static IList NormalizeDepth(IList list)
```

Comment:
The two `NormalizeDepth` methods must be consolidated as per the spreadsheet. Maybe you can add the rank as the second parameter with default argument